core/src/main/java/org/apache/brooklyn/location/byon/FixedListMachineProvisioningLocation.java

```diff
@@ -345,7 +346,7 @@ protected void updateMachineConfig(T machine, Map<?, ?> flags) {
             // For backwards compatibility, where peristed state did not have this.
             origConfigs = Maps.newLinkedHashMap();
         }
-        Map<String, Object> strFlags = ConfigBag.newInstance(flags).getAllConfig();
+        Map<String, Object> strFlags = ConfigBag.newInstanceExtending(((AbstractConfigurationSupportInternal)machine.config()).getBag(), flags).getAllConfig();
```

Comment:
Ok, didn't see that. Is this changing `FixedListMachineProvisioningLocation` to work in the same way as `JcloudsLocation`? If so, it should use `ResolvingConfigBag` for consistency with `JcloudsLocation`